### PR TITLE
Implemented the (File modification time) MDTM command.

### PR DESCRIPTION
**Pull request by hannesdejager**
See: https://tools.ietf.org/html/rfc3659#section-3.1

> The FTP command, MODIFICATION TIME (MDTM), can be used to determine
   when a file in the server NVFS was last modified.

### CODE REVIEW

File: src/server/commands.rs [hunk 1]
```diff
@@ -1305,4 +1319,25 @@ mod tests {
...
+        let tests = [
+            Test {
+                input: "MDTM\r\n",
+                expected: Err(ParseErrorKind::InvalidCommand.into()),
+            },
+            Test {
+                input: "MDTM file.txt\r\n",
+                expected: Ok(Command::MDTM { file: "file.txt".into() }),
+            },
+        ];
```
Comment (wgfm) [inline src/server/commands.rs]: Table driven tests in Rust! :D

Is this idiomatic? I like it!

Comment (hannesdejager) [reply src/server/commands.rs]: Not sure but it works.

File: src/server/mod.rs [hunk 2]
```diff
@@ -44,16 +45,13 @@ use self::stream::{SecuritySwitch, SwitchingTlsStream};
...
+use crate::storage::{self, filesystem::Filesystem, Metadata};
 use session::{Session, SessionState};
 
-use rand::Rng;
-
 const DEFAULT_GREETING: &str = "Welcome to the libunftp FTP server";
 const CONTROL_CHANNEL_ID: u8 = 0;
-
 const BIND_RETRIES: u8 = 10;
+const RFC3659_TIME: &str = "%Y%m%d%H%M%S";
```
Comment (wgfm) [inline src/server/mod.rs]: The spec mentions milliseconds, although they are optional. Any particular reason we're skipping them here?

File: src/server/mod.rs [hunk 3]
```diff
@@ -833,6 +831,25 @@ where
...
+                        Command::MDTM { file } => {
+                            ensure_authenticated!();
+                            let session = session.lock()?;
+                            let storage = Arc::clone(&session.storage);
+                            match storage.stat(&session.user, &file).wait() {
+                                Ok(meta) => match meta.modified() {
+                                    Ok(system_time) => {
+                                        let chrono_time: chrono::DateTime<chrono::offset::Utc> = system_time.into();
+                                        let formatted = chrono_time.format(RFC3659_TIME);
+                                        Ok(Reply::new(ReplyCode::FileStatus, formatted.to_string().as_str()))
```
Comment (wgfm) [inline src/server/mod.rs]: I feel like this code should have some test coverage.

File: src/server/mod.rs (hunk 2 above)
Comment (hannesdejager) [reply src/server/mod.rs]: I thought for FTP MVP its an overkill but I'm also happy to add them.

File: src/server/mod.rs (hunk 3 above)
Comment (hannesdejager) [reply src/server/mod.rs]: I feel so too I'm just not entirely sure how with the current state of the code.

File: src/server/mod.rs (hunk 2 above)
Comment (hannesdejager) [reply src/server/mod.rs]: As per discussion we'll keep it to seconds for now. ext2 & ext3 only supports seconds resolution for instance so to do this property one will have to check what the filesystem supports. For  now this is out of scope.

File: src/server/mod.rs (hunk 3 above)
Comment (hannesdejager) [reply src/server/mod.rs]: I'd like to refactor things a bit first to make it a bit more testable.

**Pull request merged**